eclair-node/src/main/scala/fr/acinq/eclair/api/serde/FormParamExtractors.scala

```diff
@@ -84,6 +84,10 @@ object FormParamExtractors {
     Offer.decode(_).get
   }
 
+  val invoiceUnmarshaller: Unmarshaller[String, Invoice] = Unmarshaller.strict { str =>
+    Bolt11Invoice.fromString(str).orElse(Bolt12Invoice.fromString(str)).get
+  }
```

Comment:
Why are you introducing this? It creates unnecessary complexity in API handlers since we don't support directly paying a Bolt 12 invoice (we should always start with an offer and go through the `invoice_request` flow)?

Reply:
Please advise how it is supposed to be done in the right way? All I need is to have an ability to parse Bolt 12 invoices  and offers.

Reply:
What I meant is that Bolt 12 invoices shouldn't be exposed, see https://github.com/ACINQ/eclair/pull/2843#discussion_r1544662941

There was a decision to not even specify a string encoding for them, we have one in eclair for consistency (and because we added it before the spec removed it), but it shouldn't be relied on.